base/range.jl

```diff
@@ -143,15 +144,15 @@ function colon{T<:AbstractFloat}(start::T, step::T, stop::T)
     if lo <= step <= hi
         a0, b = rat(start)
         a = convert(T,a0)
-        if a/convert(T,b) == start
+        if a/convert(T,b) == unitless(start)
```

Comment:
Not sure. Is there a reason you prefer it that way? I expect that what you write would also work, but if type `T` doesn't represent a number with units, is the compiler smart enough to avoid the extra division step? I've inlined a generic fallback method for `unitless` that just returns its argument, so I would expect the way I have written this now to incur no performance penalty when there are no units.